tests/deprecation/test_middleware_mixin.py

```diff
@@ -1,50 +1,105 @@
+import asyncio
 import threading
 
 from asgiref.sync import async_to_sync
 
+from django.contrib.admindocs.middleware import XViewMiddleware
+from django.contrib.auth.middleware import (
+    AuthenticationMiddleware, RemoteUserMiddleware,
+)
+from django.contrib.flatpages.middleware import FlatpageFallbackMiddleware
```

Comment:
We cannot import `FlatpageFallbackMiddleware` and `RedirectFallbackMiddleware` in the top because they import models:
```
RuntimeError: Model class django.contrib.flatpages.models.FlatPage doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.

RuntimeError: Model class django.contrib.redirects.models.Redirect doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

I will add `'deprecation'` to the `CONTRIB_TESTS_TO_APPS`.